### PR TITLE
Use Consistent types for calls

### PR DESCRIPTION
**Pull request by barry4dev**
@bluebill1049 this change seems to help - I haven't tested this fully but it gets rid of the error.  This lib needs some reworking in order to get it perfectly right, 

The problem is that you were changing types.ts without changing src/src/useForm.ts in a consistent way.  The two areas should be always referring to the same types, including the generic type.

Generally you shouldn't need the Extract<Name, string> behavior (I've honestly never used it before in my past work).  Name is based on the keyof DataType which is useful in many scenarios.

### CODE REVIEW

**Comment by bluebill1049** [comment]
thanks @barry4dev very much.

> This lib needs some reworking in order to get it perfectly right,

Would love to know what would be a good starting point?

File: src/src/useForm.ts [hunk 1]
```diff
@@ -243,14 +243,14 @@ export default function useForm<Data extends DataType>(
     validateAndStateUpdateRef.current,
   );
 
-  const setError = <Name extends keyof Data>(
-    name: Extract<Name, string>,
+  const setError = <Name extends keyof DataType>(
```
Comment (bluebill1049) [inline src/src/useForm.ts]: i think this break `useForm` types

**Comment by barry4dev** [comment]
A good starting point would be to create unit tests for both FormContext and useForm in the same branch and put it in a branch with the latest types you want to use.   Even if the tests fail for that branch it's a good single point to focus, as opposed to creating separate repo's like this.   The branch should fail to build when the types fail.

File: src/src/useForm.ts (hunk 1 above)
Comment (barry4dev) [reply src/src/useForm.ts]: Can you give an example?

Comment (bluebill1049) [reply src/src/useForm.ts]: if you do 
`const { setValue } = useForm<{ firstName: string }>();`

`setValue('firstNames', 'test');` //doesn't give a type error.

**Comment by bluebill1049** [comment]
thanks, @barry4dev this repo was just for testing purpose, going to remove it. yea I started using enzyme and was lazy to switch to react-testing-library, maybe I give it a try over the weekend.

**Comment by bluebill1049** [comment]
yea good point, I should make the build fail when types fail

**Pull request merged**